kong/clustering/wrpc_control_plane.lua

```diff
@@ -170,7 +170,7 @@ end
 
 function _M:push_config_one_client(client)
   if not self.config_call_rpc or not self.config_call_args then
-    local payload, err = self:export_deflated_reconfigure_payload()
+    local payload, err = pcall(self.export_deflated_reconfigure_payload, self)
```

Comment:
the first return value of pcall is a boolean, the original return value of `self.export_deflated_reconfigure_payload` will be the second return value with pcall. we should update the variables here to avoid confusion. since we are not using the `payload` it seems, update this to `ok, err` and L174 to `if not ok` seems ok.

Reply:
https://github.com/Kong/kong/pull/8668/commits/8718c80876cc8d9b9fc993eb3a7dc48b65d974a6

Reply:
i'm not sure exactly what error these pcalls are supposed to catch, but the fact that the tests still pass is proof to me that they don't cover the failure cases.

Reply:
i.e: an idiomatic bare lua function:

`local val, err = func(...)`
if it succeeds, `val` has a valid value
if it fails, `val` is falsy and `err` has an error message
if it errors, it doesn't return, the error is propagated upwards

the same, wrapped with pcall():

`local ok, val, err = pcall(func, ...)`
if it succeeds, `ok==true`, `val` has a valid value
if it fails, `ok==true`, `val` is falsy and `err` has an error message
if it errors, `ok==false`, `val` has an error message

so, just wrapping is not enough.  the error handling must manage the new failure paths.   This patch applies pcall() inconsistently, and doesn't handle errors.  perhaps those could be handled in a common function that does the wrapping and catches the errors correctly.

Reply:
@javierguerragiraldez Thanks for all those points... I went ahead and created a separate commit addressing your comment: https://github.com/Kong/kong/pull/8668/commits/9d0929394927b7a46cfc69c81b4190ca1646788e

Please carefully look through it... hopefully the patch is cleaner and more accurate now. I tried adding a lot of comments in the new common function that does the wrapping and the error handling to avoid confusion, but let me know if the comments are overkill or if there is anything else we need to do for this patch.

Reply:
yes, i think this is *much* better now